tests/integration/linking.rs

```diff
@@ -290,8 +290,10 @@ fn link_missing_file() {
     fs::remove_file(&out).unwrap();
 }
 
+// TODO: Ghaith please fix this :)
```

Comment:
LD linker above should take care of this as far as I can tell

Reply:
Unrelated as we decided to use LD but when using `clang` the following error is yielded:
```
Undefined symbols for architecture arm64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture arm64
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
Ok, i'll take a look soon.

The LD linker is not supposed to be able to generate an executable unless you have a correct entry point.
The current advantage of of the LD linker is that it's built in/ embedded while clang is not. 
Clang is however a compiler driver and thus makes it easier to run on different plafroms and to generate executables.
The error you are getting seems to be some missing libs in clang, not sure how that happened however.
For LD i think it compiled for the wrong target, is there a target in the test?

Reply:
> For LD i think it compiled for the wrong target, is there a target in the test?

Yes, there's a global `TARGET` variable but it's not used for that specific test.